src/v/security/tests/authorizer_test.cc

```diff
@@ -431,8 +585,18 @@ BOOST_AUTO_TEST_CASE(allow_for_all_wildcard_resource) {
 
     auth.add_bindings(bindings);
 
-    BOOST_REQUIRE(
-      auth.authorized(default_topic, acl_operation::read, user, host));
+    auto result = auth.authorized(
+      default_topic, acl_operation::read, user, host);
+
+    BOOST_REQUIRE(result.authorized);
```

Comment:
I'm not too familiar with boost test compared to gtest, but is there a way to make higher level matchers? This feels like a lot of this is copy and pasted. It makes it difficult to understand the differences.